sonic_y_cable/y_cable.py

```diff
@@ -1097,11 +1129,104 @@ def check_if_nic_lanes_active(physical_port):
 
     if platform_chassis is not None:
         res = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset, 1)
-        y_cable_validate_read_data(res, 1, physical_port, "nic lanes active")
+        if y_cable_validate_read_data(res, 1, physical_port, "nic lanes active") == EEPROM_READ_DATA_INVALID:
+            return EEPROM_ERROR
         result = res[0]
 
     else:
         helper_logger.log_error("platform_chassis is not loaded, failed to get NIC lanes active")
         return -1
 
     return result
+
+
+def get_firmware_version(physical_port, target):
+
+    data = bytearray(FIRMWARE_INFO_PAYLOAD_SIZE)
+
+    if platform_chassis is not None:
+        for byteIdx in range(0, FIRMWARE_INFO_PAYLOAD_SIZE):
+            curr_offset = 0xfc * 128 + 128 + byteIdx
+            readOut = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset, 1)
+            if y_cable_validate_read_data(readOut, 1, physical_port, "firmware info") == EEPROM_READ_DATA_INVALID:
+                return EEPROM_ERROR
+            data[byteIdx] = readOut[0]
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to get NIC lanes active")
+        return -1
+
+    result = []
```

Comment:
A list of all these values will be difficult/confusing to consume. Suggest a dict instead with descriptive keys.

Also, I wonder if we should return all of these values individually like this, or concatenate them into strings?

Reply:
changed it into dictionary. 
 strings would be better less interpretation by the caller

Reply:
> strings would be better less interpretation by the caller

Exactly. We just need to determine if the caller would ever need this granularity. If not, then we could return strings.

Reply:
fixed
sample output dict
`{'build_slot1': 'MS', 'version_slot1': '0.6', 'build_slot2': 'MS', 'version_slot2': '0.6', 'slot_status': 3}`